### PR TITLE
[#1178] Make github profile and repo hyperlinks more consistent

### LINKED ISSUE #1178: UI for viewing repo confusing
Currently we have an octocat icon to redirect users to the respective repos, as well as the same ocotcat icon directing users to github profiles when grouped by `Authors` and `merge group` selected. 

This icon is duplicated multiple times in the default setting of grouping by `Repo/Branch`, with them all pointing to the same repo.

image

This is confusing as one would think that the icon is related to the author, only to find out that they all point to the same repo.

Also, the other 2 icons changes the tab, while the octocat icon is the odd one out that redirects users to elsewhere, and is not consistent with its neighbours.

Hence, I would suggest that we make the author names and the repo names clickable to redirect users to the author profile and repo url respectively, replacing the need for the octocat icons. This would make the UI consistent no matter how we `group by` or if `merge group` is selected.

### PR DESCRIPTION
**Pull request by ccyccyccy**
Fixes #1178 and  #1191.
```
When grouped by repo, the same icon to link to the repo is duplicated
multiple times for no reason. Also, the icon to link to the author's
profile is hard to find.

We can make the links more intuitive and consistent by putting links to
the author's profile beside author names, and links to repo beside repo
names.

Let's reorder the icon group such that the hyperlinks are closer to the
respective names, as well as always showing the icons beside their
respective names. As this change heavily relies on the icons being
responsive, let's resolve issue #1191 as well to make use of
vue-fontawesome to make the icons responsive.
```

Note that there are some existing bugs that affect this PR.
1. Affected by #1185, but should work with the previews as the `config.json` is specified. Preferably merge after #1185 is resolved.

### CODE REVIEW

File: frontend/src/summary.pug [hunk 1]
```diff
@@ -154,23 +158,34 @@
             v-bind:class="{ warn: user.name === '-' }"
           ) {{ user.displayName }} ({{ user.name }})
           .summary-chart__title--contribution.mini-font [{{ user.totalCommits }} lines]
+          a(
+            v-if="filterGroupSelection !== 'groupByRepos'"
```
Comment (anubh-v) [inline frontend/src/summary.pug]: I think you're missing a comma at the end of the line, which is causing the errors from pug-lint
Here is more info on the error: [link](https://github.com/pugjs/pug-lint/blob/master/docs/rules.md#validateattributeseparator-string--object)

Comment (ccyccyccy) [reply frontend/src/summary.pug]: Oops I forgot to check the travis result. Thanks for the catch :)

**Comment by jamessspanggg** [comment]
One qn: what would be the downsides if we were to use [vue fontawesome icons](https://github.com/FortAwesome/vue-fontawesome), which alleviates the need for the span workaround?

**Comment by ccyccyccy** [comment]
> 
> 
> One qn: what would be the downsides if we were to use [vue fontawesome icons](https://github.com/FortAwesome/vue-fontawesome), which alleviates the need for the span workaround?

Depends on how the library works, but we'll need to remember to remove the useless spans after using the new library.

**Comment by jamessspanggg** [comment]
> Depends on how the library works, but we'll need to remember to remove the useless spans after using the new library.

I was thinking whether we should just switch all the icons to vue fontawesome, in this way we do not need to add and remove again. @eugenepeh @fzdy1914 may I get ur opinion on this?

**Comment by fzdy1914** [comment]
I prefer to use vue fontawesome directly.

**Comment by eugenepeh** [comment]
>I was thinking whether we should just switch all the icons to vue fontawesome, in this way we do not need to add and remove again. @eugenepeh @fzdy1914 may I get ur opinion on this?

Sounds good to me.

**Comment by ccyccyccy** [comment]
I changed to use vue-fontawesome for all the icons in the repo in this PR. Should I split it to another PR? I think it's good to merge it into this PR since this PR heavily relies on the icons being reactive.

**Comment by jylee-git** [comment]
Functionality seems to be fine.

Resolve the conflicts and request for review again

**Comment by ccyccyccy** [comment]
> Functionality seems to be fine.
> 
> Resolve the conflicts and request for review again

Fixed.

**Pull request merged**